File: frontend/src/types/alert.ts
import type { GeoPoint, Iso8601, Uuid } from './common';
import type { ThreatLevel } from './classification';

export type AlertKind =
  | 'proximity'
  | 'classification_change'
  | 'sensor_failure'
  | 'lost_track';

export interface Alert {
  id: Uuid;
  kind: AlertKind;
  threat_level: ThreatLevel;
  subject_id: Uuid | null;
  track_id: Uuid | null;
  sensor_id: Uuid | null;
  location: GeoPoint | null;
  summary: string;
  created_at: Iso8601;
  acknowledged_at: Iso8601 | null;
}

File: backend/app/domain/models/alert.py
"""Alert: an actionable notification for operators or field users."""
from __future__ import annotations

from dataclasses import dataclass
from datetime import datetime
from uuid import UUID

from app.domain.enums import AlertKind, ThreatLevel
from app.domain.models.detection import GeoPoint


@dataclass(frozen=True, slots=True)
class Alert:
    """A single alert event.

    `audience` is the set of user roles the alert should reach. The field UI
    filters more aggressively (proximity-based) than C2.
    """

    id: UUID
    kind: AlertKind
    threat_level: ThreatLevel
    subject_id: UUID | None
    track_id: UUID | None
    sensor_id: UUID | None
    location: GeoPoint | None
    summary: str
    created_at: datetime
    acknowledged_at: datetime | None
    audience: frozenset[str]
